wsmaster/che-core-api-auth-bitbucket/src/main/java/org/eclipse/che/security/oauth/BitbucketOAuthAuthenticator.java

```diff
@@ -50,35 +59,42 @@ public OAuthToken getToken(String userId) throws IOException {
     final OAuthToken token = super.getToken(userId);
     // Need to check if token is valid for requests, if valid - return it to caller.
     try {
-      if (token == null
-          || token.getToken() == null
-          || token.getToken().isEmpty()
-          || getJson(USER_URL, token.getToken(), BitbucketUser.class) == null) {
+      if (token == null || isNullOrEmpty(token.getToken())) {
         return null;
       }
+      testRequest(getTestRequestUrl(), token.getToken());
     } catch (OAuthAuthenticationException e) {
       return null;
     }
     return token;
   }
 
+  /**
+   * Generate an API request URL, to use for a token validation.
+   *
+   * @return Bitbucket Cloud or Server API request URL
+   */
+  private String getTestRequestUrl() {
+    return "https://bitbucket.org".equals(bitbucketEndpoint)
+        ? "https://api.bitbucket.org/2.0/user"
+        : bitbucketEndpoint + "/rest/api/1.0/application-properties";
+  }
```

Comment:
@vinokurig might be missing smth. but what I see from this implementation is that for `https://bitbucket.org` (SaaS) we are going to use only API 2.0, but for on-prem it will be 1.0 only (no support of 2.0)

Reply:
No, this is an API request and it is not related to oAuth 1.0 or 2.0 protocol. Bitbucket SaaS and Server have different API, so we generate a test API request according to applied SaaS or Server version.

Reply:
`1.0` and `2.0` here are just API versions, not oAuth.